src/main/java/com/github/mproberts/rxtools/map/SubjectMap.java

```diff
@@ -331,8 +331,68 @@ public void runIfBound(K key, final Runnable runnable)
         }
 
         if (isBound) {
-            runnable.run();
+            return processNewFaultForKey(key);
         }
+        return Completable.complete();
+    }
+
+    private Completable processNewFaultForKey(final K key)
+    {
+        return Completable.defer(new Callable<CompletableSource>() {
+            @Override
+            public CompletableSource call() throws Exception {
+                return new CompletableSource() {
+                    @Override
+                    public void subscribe(CompletableObserver completableObserver) {
+                        Function<K, Single<V>> faultHandler = _faultHandler;
+
+                        emitFault(key);
+
+                        if (faultHandler != null) {
```

Comment:
for indentation's sake, can you reverse this condition and make it an early return?